### DISCUSSION TITLE
Tables recognition on PDF

### DISCUSSION THREAD

**Discussion by bjmvercelli** [pymupdf/PyMuPDF#3170] [Looking for help] [answered]
Hello folks.

I've been trying to extract tables from a PDF file but i'm having this issue when there are multiple tables in the same page. In the following image you can see that it is grouping all tables as a single one...

![image](https://github.com/pymupdf/PyMuPDF/assets/61213222/f6db4bc8-19d8-49ad-ae01-7b1b84362ab4)

You can replicate it in my [Collab](https://colab.research.google.com/drive/19Ji4Ie-HEpyFx9fAfZqc0TeB6VBAjnAN?usp=sharing)

And you can find the (public) PDF here: [edital_convocacao_matricula_res__medica_2024.pdf](https://github.com/pymupdf/PyMuPDF/files/14310991/edital_convocacao_matricula_res__medica_2024.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
This is an error in the logic that recognizes clusters of vector graphics. I have developed a fix.
[Ping me if you urgently need it].

I am going to open an issue here.

**Reply by JorjMcKie** [reply]
Thank you for the feedback!
Again the question: If you have access to your PyMuPDF installation (e.g. it is on your computer), simply replace its `table.py` ... if urgent.

**Reply by bjmvercelli** [reply]
If it's not a problem for you, i would really appreciate it :)

**Reply by JorjMcKie** [reply]
no problem at all, here is the file 
[table.zip](https://github.com/pymupdf/PyMuPDF/files/14331760/table.zip)

Just unpack and copy it over to `site-packages/fitz/`, replacing the file present there. Note that there also currently still is `/fitz_old/`, the "classic" implememtnation of PyMuPDF. We are maintaining this for a while more as a fallback. It contains the same file, so you might want to do the same replacement there.

**Reply by bjmvercelli** [reply]
Looks perfect! Thank you so much. 

I'll be keeping an eye on PR's, looking foward to check on this one :trophy:

![image](https://github.com/pymupdf/PyMuPDF/assets/61213222/9d795ff2-f276-4aa4-8d5a-f56478b044fd)
